modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/RowConverter.java

```diff
@@ -108,34 +121,44 @@ public static <RowT> BinaryTuple toBinaryTuple(
 
         assert rowColumnsCount == binarySchema.elementCount() : "Invalid lookup key.";
 
-        if (IgniteUtils.assertionsEnabled()) {
-            for (int i = 0; i < rowColumnsCount; i++) {
-                if (handler.get(i, searchRow) == UNSPECIFIED_VALUE_PLACEHOLDER) {
-                    throw new AssertionError("Invalid lookup key.");
-                }
-            }
-        }
-
         BinaryTupleBuilder tupleBuilder = new BinaryTupleBuilder(rowColumnsCount);
 
-        return new BinaryTuple(rowColumnsCount, toByteBuffer(ectx, binarySchema, handler, tupleBuilder, searchRow));
+        return new BinaryTuple(rowColumnsCount, toByteBuffer(binarySchema, handler, tupleBuilder, rowColumnsCount, searchRow));
+    }
+
+    /**
+     * Converts an array of objects into a {@link ByteBuffer} representation of a {@link BinaryTuple binary tuple}. This method uses
+     * {@link RowSchema} to resolve object types.
+     *
+     * @param schema Row schema.
+     * @param values Array of objects to be stored in the buffer.
+     * @return Buffer in binary tuple format.
+     */
+    static ByteBuffer toByteBuffer(RowSchema schema, Object[] values) {
```

Comment:
> why do you put this method here 
It seemed to me that this is a more suitable place, because. this class essentially does the same RowT -> BinaryTuple conversion.

anyway, moved to SqlRowHandler.